File: Java/ARRAY/Searching.java
import java.util.Scanner;

public class Main{
    static int linearsearch(int arr[], int size, int key){
        for(int i = 0; i < size; i++){
            if(arr[i] == key){
                return i;
            }
        }
        return -1;
    }
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int[] arr = new int[]{ 1,2,3,4,5,6,7,8,9,10 };
        int size = arr.length;
        System.out.println("Enter the element you want to search: ");
        int key = in.nextInt();
        int flag = linearsearch(arr, size, key);
        if(flag == -1){
            System.out.println("Element not found");
        }
        else{
            System.out.println("Element found at: " + (flag + 1));
        }
    }
}
..................................................................
..................................................................
..................................................................
import java.util.Scanner;

public class Main{
    static int binarysearch(int arr[], int size, int key){
        int low, mid, end;
        low = 0;
        end = size - 1;
        while(low <= end){
            mid = (low + end)/2;
            if(arr[mid] == key){
                return mid;
            }
            else if (arr[mid] < key){
                low = mid + 1;
            } 
            else{
                end = mid - 1;
            }
        }
        return -1;
    }
    public static void main(String[] args){
        Scanner in = new Scanner(System.in);
        int[] arr = new int[]{ 1,2,3,4,5,6,7,8,9,10 };
        int size = arr.length;
        System.out.println("Enter the element you want to search: ");
        int key = in.nextInt();
        int flag = binarysearch(arr, size, key);
        if(flag == -1){
            System.out.println("Element not found");
        }
        else{
            System.out.println("Element found at: " + (flag + 1));
        }
    }
}

File: C++/Array/Searching.cpp
//To search for a particular element in an array.
#include<iostream>
using namespace std;

int main(){
	int arr[5];
	for(int i = 0; i < 5; i++){
		cin>>arr[i];
	}
	int key;
	cout<<"Enter the element you want to search:";
	cin>>key;
	int result;
	for(int i = 0; i<5; i++){
		if(arr[i] == key){
			result = i;
		}
	}
	cout<<"Element found at "<<result+1;
	return 0;
}
//Using function for the same thing as above:
#include<iostream>
using namespace std;

int find(int arr[], int n, int key){
	for(int i = 0; i<n;i++){
		if(key == arr[i]){
			return i;
		}
}// A function end when a value is returned. So this function end here after returning value as 'i';
	return -1;
	
}
int main(){
	int arr[] = {1,2,3,4,5};
	int n = (sizeof(arr) / sizeof(arr[0]));
	cout<<"ENter the element you want to search:";
	int key;
	cin>>key;
	int result = find(arr, n, key);
	
	if(result == -1){
		cout<<"element not found";
	}
	else{
		cout<<"Element found at "<<result+1;
	}
	return 0;
}


#include<iostream>
using namespace std;

int binarysearch(int arr[], int size, int key){
	int low, mid, high;
	low  = 0;
	high = size - 1;
	while(low <= high){
		mid = (low + high)/2;
		if(arr[mid] == key){
			return mid;
		}
		if(arr[mid < key]){
			low = mid + 1;
		}
		else{
			high = mid - 1;
		}
	}
	return -1;
}
int main(){
	int arr[] = {1,2,3,4,5,6,7,8,9,10};
	int size = sizeof(arr) / sizeof(arr[0]);
	int key = 8;
	int index = binarysearch(arr, size, key);
	cout<<"ELement found at "<<index;
	return 0;
}
